### DISCUSSION TITLE
How does messageID and sign is calculated in html curl

### DISCUSSION THREAD

**Discussion by sco0o** [krahabb/meross_lan#237] [Q&A] [answered]
Hi everyone,
I have a question. Does anyone know how the messageID and sign is called for a curl. 
I sniffed the figures and it works fine, but am not able to calculate them. I want to use this to get the right position of my cover.

`$payload = json_encode(
    array(
    'payload' => array(
        'channel' => 0),
        'header' => array(
            'messageId' => $messageId,
            'method' => 'GET',
            'from' => $url,
            'payloadVersion' => 1,
            'namespace' => 'Appliance.RollerShutter.Position',
            'uuid' => $uuid,
            'sign' => $sign,
            'triggerSrc' => 'iOSLocal',
            'timestamp' => $timestamp
    )
   )
);`

**Comment by krahabb** [comment, marked as answer]
Hello @sco0o,
the latest (alpha) release in meross_lan implements reading the device position so you might give it a try if this is the case.

At any rate, the messageId is really a simple randomly chosen GUID while the sign is a bit more comple and I don't have a clue about curl but this is a sample snippet in node.js taken from the [bytespyder's library](https://github.com/bytespider/Meross/blob/master/lib/api.js)
```
signPacket(packet) {
        const messageId = md5(uuid4())
        const timestamp = Math.floor(Date.now() / 1000)
        const signature = md5(messageId + this.key + timestamp)
        packet.header.messageId = messageId
        packet.header.timestamp = timestamp
        packet.header.sign = signature
        return packet
}
```

**Reply by sco0o** [reply]
Thanks a lot! I will take a deep dive in bytespyders library. I seems to be fit.
